projects/project-player-library/src/lib/shared/attachment-show-card/attachment-show-card.component.ts

```diff
@@ -1,24 +1,46 @@
 import { Component, EventEmitter, Input, Output } from '@angular/core';
+import { DbService } from '../../services/db/db.service';
+import { UtilsService } from '../../services/utils/utils.service';
 
 @Component({
   selector: 'lib-attachment-show-card',
   templateUrl: './attachment-show-card.component.html',
-  styleUrl: './attachment-show-card.component.css'
+  styleUrls: ['./attachment-show-card.component.css'], // Corrected the property name
 })
 export class AttachmentShowCardComponent {
-@Input() attachments:any;
-attachment:any;
-@Output() emitAttachment = new EventEmitter<any>();
-constructor(private Db:DbService){}
-ngOnInit(): void {
-  this.getAttachment();
-}
-getAttachment(){
-  this.Db.getData(this.attachments.name).then(data=>{
-    this.attachment = data.data;
-  })
-}
-actionEmit(data:any){
-  this.emitAttachment.emit(data);
-}
+  @Input() attachments: any;
+  attachment: any;
+  @Output() emitAttachment = new EventEmitter<any>();
+
+  constructor(private db: DbService,private utilService:UtilsService) {} // Renamed the variable to follow conventions
+
+  ngOnInit(): void {
+    this.getAttachment();
+  }
+
+  getAttachment() {
+    this.db.getData(this.attachments.name).then((data) => {
+      this.attachment = data.data;
+    });
+  }
+
+  actionEmit(data: any) {
+    this.emitAttachment.emit(data);
+  }
+
+  openAttachment(data: any) {
+    if(data.type === 'link'){
+      let url = data.name;
+        if (!/^https?:\/\//i.test(url)) {
```

Comment:
why this check?

Reply:
this was hitting with base api  for example local host